presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java

```diff
@@ -44,21 +51,24 @@ public InformationSchemaPageSourceProvider(Metadata metadata, AccessControl acce
     @Override
     public ConnectorPageSource createPageSource(
             ConnectorTransactionHandle transaction,
-            ConnectorSession connectorSession,
+            ConnectorSession session,
```

Comment:
the next step could be to extract prefixes (names) from table handle to several splits, so we could run this query in parallel. Just an idea to discuss.